examples/plot_simulate_lfp_array.py

```diff
@@ -0,0 +1,106 @@
+"""
+==============
+08. Record LFP
+==============
+
+This example demonstrates how to record local field potentials (LFPs).
+"""
+
+# Authors: Nick Tolley <nicholas_tolley@brown.edu>
+#          Mainak Jas <mainakjas@gmail.com>
+#          Christopher Bailey <cjb@cfin.au.dk>
+
+# sphinx_gallery_thumbnail_number = 3
+
+import os.path as op
+
+###############################################################################
+# Let us import ``hnn_core``.
+
+import hnn_core
+from hnn_core import read_params, Network, simulate_dipole
+from hnn_core.parallel_backends import MPIBackend
+
+hnn_core_root = op.dirname(hnn_core.__file__)
+
+###############################################################################
+# Then we read the parameters file
+# params_fname = op.join(hnn_core_root, 'param', 'default.json')
+params_fname = op.join(hnn_core_root, 'param', 'gamma_L5weak_L2weak.json')
+params = read_params(params_fname)
+params['tstop'] = 250
+###############################################################################
+# We will start with simulating the evoked response from
+# :ref:`evoked example <sphx_glr_auto_examples_plot_simulate_evoked.py>`.
+# We first instantiate the network.
+# (Note: Setting ``add_drives_from_params=True`` loads a set of predefined
+# drives without the drives API shown previously).
+net = Network(params)
+weights_ampa = {'L2_pyramidal': 0.0008, 'L5_pyramidal': 0.0075}
+synaptic_delays = {'L2_pyramidal': 0.1, 'L5_pyramidal': 1.0}
+rate_constant = {'L2_pyramidal': 140.0, 'L5_pyramidal': 40.0}
+net.add_poisson_drive(
+    'poisson', rate_constant=rate_constant, weights_ampa=weights_ampa,
+    location='proximal', synaptic_delays=synaptic_delays, seedcore=1079)
+
+###############################################################################
+# LFP recordings require specifying the electrode postion. It can be useful
+# to visualize the cells of the network to decide on the placement of each
+# electrode.
+net.plot_cells()
+
+###############################################################################
+# Electrode positions are stored under ``Network.lfp_array`` as a list
+# of tuples. Once we have chosen x,y,z coordinates for each electrode, we can
+# add them to the simulation.
+
+# XXX coordinates FUBAR, cortical depth direction is Y in NEURON objects!
+# electrode_pos = [(2, 400, 2), (6, 800, 6)]
+depths = list(range(-400, 1900, 200))
+electrode_pos = [(45, dep, 45) for dep in depths]
```

Comment:
Why so far from the network? Is the spiking producing pretty ridiculous voltages?

Only noticed this because I was trying some subthreshold simulations with this PR and thought I broke something when all the LFP's were flat 😄

Reply:
Do you think 45 um is far? What were the precise coordinates that produced flat LFPs? That should not happen, even very far away...

Note that this PR still includes the Y-Z-discrepancy, waiting for #314

Reply:
Perhaps flat is an exaggeration, also the same coordinates as above were used. When simulating a beta event, any oscillation is nearly imperceptible due compared to the onset transient (orders of magnitude smaller than 1 uV fluctuations). Placing the probe in the middle of the network allows for a nice visualization.

Maybe this is a good lesson though, 45 um is indeed too far _in vivo_ if you are attempting to pick up a subthreshold mechanism not in the immediate vicinity of the electrodes.

Reply:
I should play around with these myself, thanks for the pointer. By sub-threshold mechanism do you mean field oscillations? I think our simulations may not be comparable to actual measurements without accounting for a finite impedance of the electrode-tissue interface, and perhaps also the impedance of the intervening tissue itself. Both will act as an RC low pass-circuit, down-weighing the sharp spikes and emphasising slower signal components.

Reply:
my understanding was that LFP is obtained by low-pass filtering the signal and single unit activity/multi unit activity is obtained by high-pass filtering the same signal so you see spikes etc.? Perhaps this distinction should be clarified in the documentation? We could also add an argument `l_freq`, `h_freq` to the class, but maybe for a future PR?

Also, LFPy seems to have more realistic simulations, e.g., averaging over a circle so as to account for the metal contact's size (this should be like doing a low-pass filter somewhat).

Agree about fixing the issue of in-plane separation

Reply:
Perhaps the class could also be renamed to `RecordingArray` to reflect this ... because it's not LFP until you low-pass filter

Reply:
Went for `ExtracellularArray`.